engine/overworld/special_warps.asm

```diff
@@ -2,59 +2,61 @@ SpecialWarpIn::
 	call LoadSpecialWarpData
 	predef LoadTilesetHeader
 	ld hl, wd732
-	bit 2, [hl] ; dungeon warp or fly warp?
+	bit 2, [hl] ; special warp
 	res 2, [hl]
-	jr z, .next
-; if dungeon warp or fly warp
+	jr z, .debugNewGameWarp
 	ld a, [wDestinationMap]
-	jr .next2
+	jr .next
+.debugNewGameWarp
+	bit BIT_DEBUG_MODE, [hl]
+	jr z, .setNewGameMatWarp ; apply to StartNewGameDebug only
+	call PrepareNewGameDebug
+.setNewGameMatWarp
+	; This is called by OakSpeech during StartNewGame and
+	; loads the first warp event for the specified map index.
+	; Without the instruction below, the game implicitly
+	; runs "ld a, VIRIDIAN_CITY".
+	ld a, PALLET_TOWN 
 .next
-	bit 1, [hl]
-	jr z, .next3
-	call DebugStart
-.next3
-	ld a, 0
-.next2
 	ld b, a
 	ld a, [wd72d]
 	and a
-	jr nz, .next4
+	jr nz, .next2
 	ld a, b
-.next4
+.next2
 	ld hl, wd732
-	bit 4, [hl] ; dungeon warp?
+	bit 4, [hl] ; dungeon warp
 	ret nz
-; if not dungeon warp
 	ld [wLastMap], a
 	ret
 
-; gets the map ID, tile block map view pointer, tileset, and coordinates
 LoadSpecialWarpData:
 	ld a, [wd72d]
 	cp TRADE_CENTER
 	jr nz, .notTradeCenter
-	ld hl, TradeCenterSpec1
+	ld hl, TradeCenterPlayerWarp
 	ldh a, [hSerialConnectionStatus]
-	cp USING_INTERNAL_CLOCK ; which gameboy is clocking determines who is on the left and who is on the right
+	cp USING_INTERNAL_CLOCK ; determine player and friend
 	jr z, .copyWarpData
-	ld hl, TradeCenterSpec2
+	ld hl, TradeCenterFriendWarp
 	jr .copyWarpData
 .notTradeCenter
 	cp COLOSSEUM
 	jr nz, .notColosseum
-	ld hl, ColosseumSpec1
+	ld hl, ColosseumPlayerWarp
 	ldh a, [hSerialConnectionStatus]
-	cp USING_INTERNAL_CLOCK
+	cp USING_INTERNAL_CLOCK ; determine player and friend
 	jr z, .copyWarpData
-	ld hl, ColosseumSpec2
+	ld hl, ColosseumFriendWarp
 	jr .copyWarpData
 .notColosseum
 	ld a, [wd732]
-	bit 1, a
-	jr nz, .notFirstMap
-	bit 2, a
-	jr nz, .notFirstMap
-	ld hl, FirstMapSpec
+	bit BIT_DEBUG_MODE, a
+	; warp to wLastMap (PALLET_TOWN) for StartNewGameDebug
+	jr nz, .notNewGameWarp
+	bit 2, a ; special warp
```

Comment:
Please don't add *more* comments which should be constants, if you identified that bit 2 is for special warp then why comment every such line instead of defining `SPECIAL_WARP_BIT`?

Reply:
Likewise, let's do this in a dedicated PR instead.

Reply:
Agreed.